contract/vm.go

```diff
@@ -238,40 +238,26 @@ func NewVmContextQuery(
 	return ctx, nil
 }
 
-func (ctx *vmContext) IsGasSystem() bool {
-	return !ctx.isQuery && PubNet && ctx.blockInfo.ForkVersion >= 2
-}
-
 // get the remaining gas from the given LState
 func (ctx *vmContext) refreshRemainingGas(L *LState) {
-	if ctx.IsGasSystem() {
+	if fee.IsVmGasSystem(ctx.blockInfo.ForkVersion, ctx.isQuery) {
```

Comment:
The same here. The original line is simpler

Reply:
it is same. fee calculation logic is separated in ctx, so it need to extract args in ctx.
but, as you said, current code is not simple.